lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/taxi/TaxiVehicleData.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2025 Fraunhofer FOKUS and others. All rights reserved.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contact: mosaic@fokus.fraunhofer.de
+ */
+
+package org.eclipse.mosaic.lib.objects.taxi;
+
+import org.eclipse.mosaic.lib.objects.vehicle.VehicleData;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import java.io.Serial;
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * Holds information about taxi/drt vehicles, such as their
+ * transport capacity and service status.
+ */
+public class TaxiVehicleData implements Serializable {
+
+    @Serial
+    private static final long serialVersionUID = 1L;
+
+    public enum TaxiState {
+        EMPTY, PICKUP, OCCUPIED, OCCUPIED_AND_PICK_UP;
+
+        public static TaxiState of(int stateIdSumo) {
+            return switch (stateIdSumo) {
+                case 0 -> EMPTY;
+                case 1 -> PICKUP;
+                case 2 -> OCCUPIED;
+                case 3 -> OCCUPIED_AND_PICK_UP;
+                default -> throw new IllegalArgumentException("Unknown state: " + stateIdSumo);
+            };
+        }
+
+    }
+
+    private final String id;
+    private final TaxiState state;
+    private final int personCapacity;
+    private final VehicleData vehicleData;
```

Comment:
This seems a bit confusing. `VehicleData` already holds an ID, is this different from the id field.
Why doesn't the `TaxiVehicleData` extend the `VehicleData`?
Question to myself: What about electric taxis then?